src/Microsoft.ML.Data/DataView/RowToRowMapperTransform.cs

```diff
@@ -88,7 +88,12 @@ private static VersionInfo GetVersionInfo()
 
         public override DataViewSchema OutputSchema => _bindings.Schema;
 
-        bool ICanSaveOnnx.CanSaveOnnx(OnnxContext ctx) => _mapper is ICanSaveOnnx onnxMapper ? onnxMapper.CanSaveOnnx(ctx) : false;
+        bool ICanSaveOnnx.CanSaveOnnx(OnnxContext ctx)
+        {
+            const int minimumOpSetVersion = 9;
+            Contracts.Assert(ctx.GetOpSetVersion() >= minimumOpSetVersion, "OpSet version " + ctx.GetOpSetVersion() + " is older than " + LoaderSignature + "'s minimum OpSet version requirement: " + minimumOpSetVersion);
+            return _mapper is ICanSaveOnnx onnxMapper ? onnxMapper.CanSaveOnnx(ctx) : false;
```

Comment:
This shouldn't use Contracts.Assert since it will fire only during Debug builds.  Contracts.CheckValue would work better

Reply:
From my end I see what I use is Contracts.ExceptParam(), perhaps that's the github's latency. Yeah I think that CheckValue() is better

Reply:
I then found the checkvalue() is not what I expected(comparing with value instead of null). So I feel it's better in this case to use throw ExceptParam()

Reply:
There is also `CheckParam()` where you give it the condition and the error messages, it's a bit shorter than writing the `if` clause. Although, if the error message string needs to be formatted, we usually use `if (!cond) throw Contracts.Except...` so that we don't actually format the string unless the exception is thrown.

---
In reply to: [432769528](https://github.com/dotnet/machinelearning/pull/5176#discussion_r432769528) [](ancestors = 432769528)